scanner/src/lib/dataproducts/LogisticsEmissions_v0_1.svelte

```diff
@@ -50,9 +50,106 @@
     seaFreightEmissions: Array<SeaFreightEmissions>
     waybillNumber: string
   }
+
+  const totalEmissionsRoadFreight = data.roadFreightEmissions.reduce(
+    (acc, roadFreightEmission) => acc + roadFreightEmission.totalEmissions,
+    0
+  )
+  const totalEmissionsSeaFreight = data.seaFreightEmissions.reduce(
+    (acc, seaFreightEmission) => acc + seaFreightEmission.totalEmissions,
+    0
+  )
+  const totalEmissions = totalEmissionsRoadFreight + totalEmissionsSeaFreight
+
+  const totalEmissionsIntensityRoadFreight = data.roadFreightEmissions.reduce(
+    (acc, roadFreightEmission) => acc + roadFreightEmission.emissionIntensity,
+    0
+  )
+  const totalEmissionsIntensitySeaFreight = data.seaFreightEmissions.reduce(
+    (acc, seaFreightEmission) => acc + seaFreightEmission.emissionIntensity,
+    0
+  )
+  const totalEmissionsIntensity =
+    totalEmissionsIntensityRoadFreight + totalEmissionsIntensitySeaFreight
+
+  const averageEmissionsIntensityRoadFreight =
+    totalEmissionsIntensityRoadFreight / data.roadFreightEmissions.length
+  const averageEmissionsIntensitySeaFreight =
+    totalEmissionsIntensitySeaFreight / data.seaFreightEmissions.length
+  function averageEmissionsIntensity() {
+    if (isNaN(averageEmissionsIntensityRoadFreight)) {
+      if (isNaN(averageEmissionsIntensitySeaFreight)) {
+        return 0
+      } else {
+        return averageEmissionsIntensitySeaFreight
+      }
+    } else {
+      if (isNaN(averageEmissionsIntensitySeaFreight)) {
+        return averageEmissionsIntensityRoadFreight
+      } else {
+        return (averageEmissionsIntensityRoadFreight + averageEmissionsIntensitySeaFreight) / 2
+      }
+    }
+  }
```

Comment:
Even if still wrong this would be better as (approx)
```
const avg =  (totalRoad + totalSea) / (roadLength + seaLength)
```
now you're averaging averages incorrectly .. but the entire thing is very incorrect anyway and that part we cannot fix here :smile:

Reply:
generally by moving some of the steps to variables you should be able to simplify this anyway